src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs

```diff
@@ -0,0 +1,469 @@
+        [InlineData(MemoryMappedFileAccess.Read, MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(MemoryMappedFileAccess.Read, MemoryMappedFileAccess.ReadExecute)]
+        [InlineData(MemoryMappedFileAccess.Read, MemoryMappedFileAccess.ReadWriteExecute)]
+        public void InvalidAccessLevelsCombinations(MemoryMappedFileAccess mapAccess, MemoryMappedFileAccess viewAccess)
+        {
+            const int Capacity = 4096;
+            using (MemoryMappedFile mmf = MemoryMappedFile.CreateNew(null, Capacity, mapAccess))
+            {
+                Assert.Throws<UnauthorizedAccessException>(() => mmf.CreateViewAccessor(0, Capacity, viewAccess));
+            }
+        }
+
+        /// <summary>
+        /// Test to verify the accessor's PointerOffset.
+        /// </summary>
+        [Fact]
+        public void PointerOffsetMatchesViewStart()
+        {
+            const int MapLength = 4096;
+            foreach (MemoryMappedFile mmf in CreateSampleMaps(MapLength))
+            {
+                using (mmf)
+                {
+                    using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor())
+                    {
+                        Assert.Equal(0, acc.PointerOffset);
+                    }
+
+                    using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor(0, MapLength))
+                    {
+                        Assert.Equal(0, acc.PointerOffset);
+                    }
+                    using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor(1, MapLength - 1))
+                    {
+                        Assert.Equal(1, acc.PointerOffset);
+                    }
+                    using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor(MapLength - 1, 1))
+                    {
+                        Assert.Equal(MapLength - 1, acc.PointerOffset);
+                    }
+
+                    // On Unix creating a view of size zero will result in an offset and capacity
+                    // of 0 due to mmap behavior, whereas on Windows it's possible to create a
+                    // zero-size view anywhere in the created file mapping.
+                    using (MemoryMappedViewAccessor acc = mmf.CreateViewAccessor(MapLength, 0))
+                    {
+                        Assert.Equal(
+                            Interop.IsWindows ? MapLength : 0,
+                            acc.PointerOffset);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Test all of the Read/Write accessor methods against a variety of maps and accessors.
+        /// </summary>
+        [Theory]
+        [InlineData(0, 8192)]
+        [InlineData(8100, 92)]
+        [InlineData(0, 20)]
+        [InlineData(1, 8191)]
+        [InlineData(17, 8175)]
+        [InlineData(17, 20)]
+        public void AllReadWriteMethods(long offset, long size)
+        {
+            foreach (MemoryMappedFile mmf in CreateSampleMaps(8192))
```

Comment:
Good question, but I actually wanted this to be 8192, a value at least as large as any of the offset+size combinations.  I could have also made it offset+size, but part of what I wanted tested was creating accessors that are small slices of the overall map.